frontend/packages/dev-console/src/utils/__tests__/add-resources-menu-utils.spec.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { URL } from 'url';
+import { GitAltIcon } from '@patternfly/react-icons';
+import { getMenuPath, getAddPageUrl, createKebabAction } from '../add-resources-menu-utils';
+import {
+  transformTopologyData,
+  getTopologyResourceObject,
+} from '../../components/topology/topology-utils';
+import { ImportOptions } from '../../components/import/import-types';
+import { MockResources } from '../../components/topology/__tests__/topology-test-data';
+import { TopologyDataResources } from '../../components/topology/topology-types';
+
+expect.extend({
+  toHaveProperties(received, argument) {
+    const pass = argument.every((arg: string) => {
+      return _.has(received, arg);
+    });
+
+    if (pass) {
+      return {
+        message: () =>
+          `expected  ${this.utils.printReceived(
+            received,
+          )} to contain Keys ${this.utils.printExpected(argument)}`,
+        pass: true,
+      };
+    }
+    return {
+      message: () =>
+        `expected ${this.utils.printReceived(received)} to contain Keys ${this.utils.printExpected(
+          argument,
+        )} but got ${this.utils.printExpected(Object.keys(received))}`,
+      pass: false,
+    };
+  },
+});
+
+const getTopologyData = (mockData: TopologyDataResources, transformByProp: string[]) => {
+  const result = transformTopologyData(mockData, transformByProp);
+  const keys = Object.keys(result.topology);
+  const resource = getTopologyResourceObject(result.topology[keys[0]]);
+  return { resource };
+};
+
+describe('addResourceMenuUtils: ', () => {
+  it('should give proper menu item path based on the application', () => {
+    expect(getMenuPath(true)).toEqual('Add to Application');
+    expect(getMenuPath(false)).toEqual('Add to Project');
+  });
+
+  it('should return the page url with proper queryparams for git import flow', () => {
+    const { resource } = getTopologyData(MockResources, ['deployments']);
+    const url = new URL(getAddPageUrl(resource, ImportOptions.GIT, true), 'https://mock.test.com');
+    // Check that url contains all expected params
```

Comment:
Since you have to make a change already, can you remove all these comments as they don't attribute themselves to the test structure anymore.